apps/web/pages/api/auth/forgot-password.ts

```diff
@@ -1,14 +1,31 @@
 import type { ResetPasswordRequest } from "@prisma/client";
 import type { NextApiRequest, NextApiResponse } from "next";
+import { z } from "zod";
 
 import dayjs from "@calcom/dayjs";
 import { sendPasswordResetEmail } from "@calcom/emails";
 import { PASSWORD_RESET_EXPIRY_HOURS } from "@calcom/emails/templates/forgot-password-email";
+import rateLimit from "@calcom/lib/rateLimit";
 import { getTranslation } from "@calcom/lib/server/i18n";
 import prisma from "@calcom/prisma";
 
+const limiter = rateLimit({
```

Comment:
We need to come up with a better strategy. As discussed with @roae `memory-cache` doesn't play nice with server less environments.

Reply:
Doesn't have to be perfect though for rate limiting, whether this needs fixing needs to be shown by real world tests, perhaps it isn't exactly 10 times but if it ratelimits after 50 in different serverless runners that's still achieving the necessary limiting.